src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,26 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+; TODO: actually infer the bucket type
+(defn infer-bucket-type
+  "infer bucket type from bucket name"
+  [bucket-name]
+  (if (.startsWith bucket-name "gs://") :gs :s3))
```

Comment:
So `starts-with?` is only available from clojure 1.8. I have updated `project.clj` to use clojure 1.8 instead of 1.7 and changed the function to use `starts-with?`. Tests are still parsing but let me know if this could be an issue?